Makefile

```diff
@@ -243,6 +243,9 @@ abyss: debug
 format:
 	clang-format -i $$(find jni daemon llarp include libabyss | grep -E '\.[h,c](pp)?$$')
 
+format-verify: format
+	git diff --quiet || echo "Please run make format!!" && git --no-pager diff && false
```

Comment:
shouldn't this be `git format --quiet` ? or something?

Reply:
Not sure what you mean:

```
$ git format --help
git: 'format' is not a git command. See 'git --help'.
```

what this PR essentially does:

- run the normal `make format` check
- run `git diff`. If there are any changes and there shouldn't be, because we're perfect and never forget to run `make format`, and travis does a clean checkout.
- print the diff if there are any changes